src/ring_theory/ideals.lean

```diff
@@ -70,7 +70,10 @@ span_le.trans $ singleton_subset_iff.trans mem_span_singleton
 
 lemma span_eq_bot {s : set α} : span s = ⊥ ↔ ∀ x ∈ s, (x:α) = 0 := submodule.span_eq_bot
 
-lemma span_singleton_eq_bot {x} : span ({x} : set α) = ⊥ ↔ x = 0 := submodule.span_singleton_eq_bot
+@[simp] lemma span_singleton_eq_bot {x} : span ({x} : set α) = ⊥ ↔ x = 0 :=
+submodule.span_singleton_eq_bot
+
+@[simp] lemma span_zero : span (0 : set α) = ⊥ := by rw [←set.singleton_zero, span_singleton_eq_bot]
```

Comment:
I changed this from `span {0}` to `span 0` because of the linter [here](https://github.com/leanprover-community/mathlib/runs/861272279#step:5:41).

Oddly enough the other linter warning disappeared after the merge.

Reply:
[Oops, I guess not](https://github.com/leanprover-community/mathlib/pull/3306/checks?check_run_id=886298276#step:5:26). It didn't show up when I put `#lint` at the bottom of `linear_algebra/basic.lean`...

Reply:
... and the reason was because `set α` gets its `has_zero` instance in `algebra.pointwise`, which wasn't previously imported in `linear_algebra.basic`. But now it is.